tools/workspace/new_release.py

```diff
@@ -392,10 +415,54 @@ def main():
         print(json.dumps(metadata, sort_keys=True, indent=2))
 
     if workspaces is not None:
+        # Generate a set of workspaces which are members of a cohort and
```

Comment:
I'm not really following this logic, especially the first new block. I wonder if it wouldn't make more sense to just identify if a listed workspace is part of a cohort, and, if so, pass the cohort to `_do_upgrade` (which would allow the user to specify only one of a cohort) and let that deal with whether or not it is able to commit.

Something like:
```python
        visited_workspaces = set()
        for workspace in workspaces:
            # If we already did this as part of a tandem upgrade, skip it.
            if workspace in visited_workspaces:
                continue

            # Determine if this workspace needs to be a tandem upgrade.
            cohort_workspaces = {workspace}
            for cohort in _COHORTS:
                if workspace in cohort:
                    cohort_workspaces = tandem_workspace_set

            # Actually do the upgrade(s).
            with TemporaryDirectory(prefix='drake_new_release_') as temp_dir:
                _do_upgrades(temp_dir, gh, local_drake_checkout,
                             cohort_workspaces, metadata)
                visited_workspaces.update(cohort_workspaces)
```